### ISSUE TITLE
Copy Pasting of multiple images from one WYSIWYG editor to Froala not working

### ISSUE THREAD

**Issue by ShanmugaPriyanV** [froala/wysiwyg-editor#2246]
##### Expected behavior.
Copy Pasting of multiple images from textAngular (http://textangular.com/) to Froala Editor should work after applying the solution from https://github.com/froala/wysiwyg-editor/issues/2232
##### Actual behavior.
Copy Pasting of multiple images from textAngular (http://textangular.com/) to Froala Editor is happening in a weird way. When i copy two images from textAngular to Froala either single image gets repeated or one of the image gets repeated twice resulting in three images.

##### Steps to reproduce the problem.
1) Open http://textangular.com/ and insert image by drag&drop and copy the entire content from textAngular.
2) Open jsfiddle https://jsfiddle.net/80b7mvv9/3/ (this has the fix from https://github.com/froala/wysiwyg-editor/issues/2232) and paste the copied content.

##### OS.
Windows 7

##### Browser.
Chrome Version 61.0.3163.100 (Official Build) (32-bit)

Image copied from textAngular
![image](https://user-images.githubusercontent.com/20870994/31213778-d98a198e-a9c5-11e7-83c9-d5eaf9330179.png)

Image pasted in Froala and the Result
![image](https://user-images.githubusercontent.com/20870994/31213787-e4f17b0a-a9c5-11e7-89a2-33ba3cfa11d5.png)

Thanks in Advance

**Comment by stefanneculai** [comment]
@ShanmugaPriyanV could you please provide a link to a page from where we can copy directly with the images? Thank you.

**Comment by stefanneculai** [comment]
We've just checked and appears to work just fine. Would be happy to check it if you could provide the exact steps to reproduce the issue. Thank you in advance!

**Comment by ansh-les** [comment]
I just ran into the same issue. While pasting multiple images, the first image gets replicated in place of the last image.
I was using the `image.beforeUpload` event to convert the images into base64 as mentioned here:
https://wysiwyg-editor.froala.help/hc/en-us/articles/115000555949-Can-I-insert-images-as-base64

What I copied:
![image](https://user-images.githubusercontent.com/57872180/100847608-0a7b4b80-34a6-11eb-97f1-2590575df521.png)

What got pasted:
![image](https://user-images.githubusercontent.com/57872180/100847645-16670d80-34a6-11eb-82c1-f3d880cf3a3f.png)

### Solution
My understanding of this issue is that the `editor.image.get()` was being evaluated inside `reader.onload` which is called AFTER the image is processed (which takes time). But by that time, editor is processing a different image so `editor.image.get()` evaluates to a different image.

So, I just stored the current image in a variable and used that inside in the `onload` callback.
```js
"image.beforeUpload": function (files) {
  const editor = this

  if (files.length) {
    const currentImage = editor.image.get()
    
    // Create a File Reader.
    const reader = new FileReader()

    // Set the reader to insert images when they are loaded.
    reader.onload = function (e) {
      const result = e.target.result
      // console.log(result)
      editor.image.insert(result, null, null, currentImage)
    }

    // Read image as base64.
    reader.readAsDataURL(files[0])
  }

  editor.popups.hideAll()

  // Stop default upload chain.
  return false
},
```

**Comment by HemangiModi13** [comment]
I have resolved this problem by downgrading to version 3.1.0 from 3.2.0
